src/aggregation/bucket/range.rs

```diff
@@ -0,0 +1,546 @@
+        if val == u64::MIN || val == u64::MAX {
+            "*".to_string()
+        } else {
+            f64_from_fastfield_u64(val, field_type).to_string()
+        }
+    };
+
+    format!("{}-{}", to_str(range.start), to_str(range.end))
+}
+
+pub fn range_to_key(range: &Range<u64>, field_type: &Type) -> Key {
+    Key::Str(range_to_string(range, field_type))
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn bucket_test_extend_range_hole() {
+        let buckets = vec![(10f64..20f64).into(), (30f64..40f64).into()];
+        let buckets = extend_validate_ranges(&buckets, &Type::U64).unwrap();
+        assert_eq!(buckets[0].start, u64::MIN);
+        assert_eq!(buckets[0].end, 10);
+        assert_eq!(buckets[1].start, 10);
+        assert_eq!(buckets[1].end, 20);
+        // Added bucket to fill hole
+        assert_eq!(buckets[2].start, 20);
+        assert_eq!(buckets[2].end, 30);
+        assert_eq!(buckets[3].start, 30);
+        assert_eq!(buckets[3].end, 40);
+    }
+
+    #[test]
+    fn bucket_range_test_negative_vals() {
+        let buckets = vec![(-10f64..-1f64).into()];
+        let buckets = extend_validate_ranges(&buckets, &Type::F64)
+            .unwrap()
+            .iter()
+            .map(|bucket| range_to_string(bucket, &Type::F64))
+            .collect_vec();
+        assert_eq!(buckets[0], "*--10");
+        assert_eq!(buckets[buckets.len() - 1], "-1-*");
+    }
+    #[test]
+    fn bucket_range_test_positive_vals() {
+        let buckets = vec![(0f64..10f64).into()];
+        let buckets = extend_validate_ranges(&buckets, &Type::F64)
+            .unwrap()
+            .iter()
+            .map(|bucket| range_to_string(bucket, &Type::F64))
+            .collect_vec();
+        assert_eq!(buckets[0], "*-0");
+        assert_eq!(buckets[buckets.len() - 1], "10-*");
+    }
+
+    #[test]
+    fn range_binary_search_test() {
+        let ranges = vec![(u64::MIN..10), (10..100), (100..u64::MAX)];
+
+        let search = |val: u64| {
+            ranges
+                .binary_search_by(|range| match range.contains(&val) {
+                    true => Ordering::Equal,
+                    false => {
+                        if range.end == val {
+                            Ordering::Less
+                        } else {
+                            range.end.cmp(&val)
+                        }
+                    }
+                })
+                .unwrap_or_else(|val| val - 1) // U64::MAX case
+        };
+
+        assert_eq!(search(u64::MIN), 0);
+        assert_eq!(search(9), 0);
+        assert_eq!(search(10), 1);
+        assert_eq!(search(11), 1);
+        assert_eq!(search(99), 1);
+        assert_eq!(search(100), 2);
+        assert_eq!(search(u64::MAX), 2);
+    }
+}
+
+#[cfg(all(test, feature = "unstable"))]
+mod bench {
+
+    use std::cmp::Ordering;
+
+    use rand::seq::SliceRandom;
+    use rand::thread_rng;
+
+    use super::*;
+
+    #[derive(Clone)]
```

Comment:
These bench have been super helpful, thanks.

They are however disconnected from the code.
Could you remove them and add one bench that actually use the path code?

Ideally that bench should be an util function that take the number of buckets as a parameter so we can have a few bucket and a many bucket bench.

Reply:
uses now actual code and is parameterized